lib/ansible/modules/cloud/azure/azure_rm_resource.py

```diff
@@ -286,7 +286,7 @@ def exec_module(self, **kwargs):
         needs_update = True
         response = None
 
-        if self.idempotency:
+        if self.idempotency or self.method == "DELETE":
```

Comment:
self.idempotency always FALSE. what's the point of it?

Reply:
idempotency option is just to enable idempotency check in present state, so i think it shouldn't be necessary to use this option when deleting resource.